### ISSUE TITLE
Rancher - It is not possible to create Node/Clusters using RKE2/K3s, only RKE1 is available

### ISSUE THREAD

**Issue by BigBrather** [JonasProgrammer/docker-machine-driver-hetzner#118]
I can't create a Kubernetes Cluster on Hetzner Cloud using RKE2/K3s.

Here's an example, when I have RKE1 selected, I can create a Kubernetes Cluster:
![image](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/99731862/54040ec6-8dc9-4336-b303-d9e935eec91e)

and

![image](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/99731862/eedf4dd2-c248-4a82-aa73-31e83eaff0a2)

But when I select RKE2/K3s, the image on the Hetzner driver disappears:

![image](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/99731862/46ef48d1-2589-4369-8425-75b5e9d28f22)

After selecting Hetzner Cloud, I click continue, where I get to the next window, in which I save the values ​​​​and click continue, and I get into a loop, I add a variable through the file and when continuing, it says to add credentials again:

![image](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/99731862/e9b9ee3c-e453-4ad6-865f-e04afba709c4)

I have a question for the GitHub audience:
1. Has anyone encountered this problem?

And also have a question for the driver developers:
1. As I understand it, the problem here is the support for RKE2/K3s, is there a chance that it will appear for this driver in this bundle, like mine?

I am using Rancher v2.7.9 and Driver Hetzner v5.0.1.

**Comment by gn-mtrlitza** [comment]
Hi there.
I have the same behavior with rancher v2.8.0 and v5.0.2 of Hetzner driver. Looking forward to an answer to @BigBrather's questions 🙂

**Comment by mihalsoft** [comment]
Hi,
I had the same issue. The problem was, that I had to add **Cloud credentials** firstly.
![Screenshot 2024-01-29 at 15 49 04](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/56034988/d5ae2908-b1fd-42ec-8114-b4afe59a67ec)

After installing Hetzner driver, I didn't see Hetzner option for adding Cloud credential. I just did **hard refresh of page** and the Hetzner option has appeard.

Rancher version v.2.7.9 and v5.0.2 of Hetzner driver.

In short:
1. Install Hetzner driver.
2. Go to the Cloud credentials -> Create -> Select Hetzner (If you can't see hetzner, try to hard refresh page)
3. Go to Clusters -> Create ->  Select Hetzner (RKE2) [There is something wrong with logo image]
4. Click create cluster

**Comment by BigBrather** [comment]
> Hi, I had the same issue. The problem was, that I had to add **Cloud credentials** firstly. ![Screenshot 2024-01-29 at 15 49 04](https://private-user-images.githubusercontent.com/56034988/300501495-d5ae2908-b1fd-42ec-8114-b4afe59a67ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcxMjY5NDEsIm5iZiI6MTcwNzEyNjY0MSwicGF0aCI6Ii81NjAzNDk4OC8zMDA1MDE0OTUtZDVhZTI5MDgtYjFmZC00MmVjLTgxMTQtYjRhZmU1OWE2N2VjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA1VDA5NTA0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2ZTMxNzM4M2NkMDdmYzE4ODhhZmI0YmZkNzJlNTdlNmJmZDJlOThkOWRmNDAyMTAxYzExZDhhNzhjNmFjMTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Axu_qgn_3mq-WqCjDrVm3rqInDXiMjKuJdJC_xKJsCk)
> 
> After installing Hetzner driver, I didn't see Hetzner option for adding Cloud credential. I just did **hard refresh of page** and the Hetzner option has appeard.
> 
> Rancher version v.2.7.9 and v5.0.2 of Hetzner driver.
> 
> In short:
> 
> 1. Install Hetzner driver.
> 2. Go to the Cloud credentials -> Create -> Select Hetzner (If you can't see hetzner, try to hard refresh page)
> 3. Go to Clusters -> Create ->  Select Hetzner (RKE2) [There is something wrong with logo image]
> 4. Click create cluster

It doesn't work for me, same versions as you Hetzen does not appear when updating.

![image](https://github.com/JonasProgrammer/docker-machine-driver-hetzner/assets/99731862/ba9ae83c-48fe-4e80-9b14-c1c37eb30952)

**Comment by mihalsoft** [comment]
Try to install Hetzner Node driver using Local cluster not UI.

Steps:
1. Uninstall existing Hetzner driver if it's already installed
1. Go to Local cluster using Rancher
2. Click import yaml in top right corner
3. Paste and click import:

``` yaml
apiVersion: management.cattle.io/v3
kind: NodeDriver
metadata:
   annotations:
      lifecycle.cattle.io/create.node-driver-controller: "true"
      privateCredentialFields: apiToken
   generation: 13
   name: hetzner
spec:
   active: true
   addCloudCredential: false
   builtin: false
   checksum: ""
   description: ""
   displayName: hetzner
   externalId: ""
   uiUrl: https://storage.googleapis.com/hcloud-rancher-v2-ui-driver/component.js
   url: https://github.com/JonasProgrammer/docker-machine-driver-hetzner/releases/download/5.0.2/docker-machine-driver-hetzner_5.0.2_linux_amd64.tar.gz
   whitelistDomains:
   - storage.googleapis.com
```

**Comment by bartoszpijet** [comment]
> Try to install Hetzner Node driver using Local cluster not UI.
> 
> Steps:
> 
>     1. Uninstall existing Hetzner driver if it's already installed
> 
>     2. Go to Local cluster using Rancher
> 
>     3. Click import yaml in top right corner
> 
>     4. Paste and click import:
> 
> 
> ```yaml
> apiVersion: management.cattle.io/v3
> kind: NodeDriver
> metadata:
>    annotations:
>       lifecycle.cattle.io/create.node-driver-controller: "true"
>       privateCredentialFields: apiToken
>    generation: 13
>    name: hetzner
> spec:
>    active: true
>    addCloudCredential: false
>    builtin: false
>    checksum: ""
>    description: ""
>    displayName: hetzner
>    externalId: ""
>    uiUrl: https://storage.googleapis.com/hcloud-rancher-v2-ui-driver/component.js
>    url: https://github.com/JonasProgrammer/docker-machine-driver-hetzner/releases/download/5.0.2/docker-machine-driver-hetzner_5.0.2_linux_amd64.tar.gz
>    whitelistDomains:
>    - storage.googleapis.com
> ```

Works great, thanks for providing this solution!

**Comment by BigBrather** [comment]
> Try to install Hetzner Node driver using Local cluster not UI.
> 
> Steps:
> 
> 1. Uninstall existing Hetzner driver if it's already installed
> 2. Go to Local cluster using Rancher
> 3. Click import yaml in top right corner
> 4. Paste and click import:
> 
> ```yaml
> apiVersion: management.cattle.io/v3
> kind: NodeDriver
> metadata:
>    annotations:
>       lifecycle.cattle.io/create.node-driver-controller: "true"
>       privateCredentialFields: apiToken
>    generation: 13
>    name: hetzner
> spec:
>    active: true
>    addCloudCredential: false
>    builtin: false
>    checksum: ""
>    description: ""
>    displayName: hetzner
>    externalId: ""
>    uiUrl: https://storage.googleapis.com/hcloud-rancher-v2-ui-driver/component.js
>    url: https://github.com/JonasProgrammer/docker-machine-driver-hetzner/releases/download/5.0.2/docker-machine-driver-hetzner_5.0.2_linux_amd64.tar.gz
>    whitelistDomains:
>    - storage.googleapis.com
> ```

Your solution works for me, now I can create RKE1 and RKE2. Thank you!
